lib/rouge/lexers/kotlin.rb

```diff
@@ -56,6 +56,18 @@ class Kotlin < RegexLexer
           groups Keyword::Declaration, Text
           push :property
         end
+        rule %r'(fun)(\s+)(<)(#{name})(>)(\s+)(#{name})(\.)' do
```

Comment:
It's too simple.

```kotlin
fun <T, U> f() {}
fun <T>f() { }
```

Don't use long regexp to parse nestable syntax parts.

Reply:
@gfx Any suggestions how to make it right? It was pretty challenging to get it even this far. May be you can point me to an example where something like this was implemented?